### ISSUE TITLE
Mirroring breaks if one of the repos has malformed metadata

### ISSUE THREAD

**Issue by ikapelyukhin** [SUSE/rmt#97]
I.e. if HTML is returned instead of `directory.yast` in one of the repos, that breaks the whole mirroring process:

```
I, [2018-02-12T12:32:16.925724 #22884]  INFO -- : ↓ directory.yast
~/.rvm/rubies/ruby-2.4.1/lib64/ruby/2.4.0/uri/rfc3986_parser.rb:67:in `split': bad URI(is not URI?): <!DOCTYPE html> (URI::InvalidURIError)
	from ~/.rvm/rubies/ruby-2.4.1/lib64/ruby/2.4.0/uri/rfc3986_parser.rb:73:in `parse'
	from ~/.rvm/rubies/ruby-2.4.1/lib64/ruby/2.4.0/uri/rfc3986_parser.rb:117:in `convert_to_uri'
	from ~/.rvm/rubies/ruby-2.4.1/lib64/ruby/2.4.0/uri/generic.rb:1099:in `merge'
	from ~/.rvm/rubies/ruby-2.4.1/lib64/ruby/2.4.0/uri/rfc3986_parser.rb:89:in `inject'
	from ~/.rvm/rubies/ruby-2.4.1/lib64/ruby/2.4.0/uri/rfc3986_parser.rb:89:in `join'
	from ~/.rvm/rubies/ruby-2.4.1/lib64/ruby/2.4.0/uri/common.rb:269:in `join'
	from ~/work/rmt/lib/rmt/downloader.rb:98:in `make_request'
	from ~/work/rmt/lib/rmt/downloader.rb:38:in `block in download'
```

Ideally, the affected repo should be skipped and mirroring would continue.

**Comment by kalabiyau** [comment]
@ikapelyukhin could you please check if that file is listed by repomd.xml and send me that repository somehow. I believe we could ensure that repos are valid at the time we publish the product in SCC.

**Comment by ikapelyukhin** [comment]
@kalabiyau in my case the issue was caused by a custom repo.

**Comment by thutterer** [comment]
@kalabiyau This also happens when one repo has an invalid checksum. Saw this happening on reference host:

```
I, [2018-06-12T14:07:52.548590 #25142]  INFO -- : ↓ gnome-nibbles-3.24.0-3.1.x86_64.rpm
Traceback (most recent call last):
	2: from /usr/share/rmt/lib/rmt/downloader.rb:86:in `block in process_queue'
	1: from /usr/share/rmt/lib/rmt/downloader.rb:134:in `finalize_download'
/usr/share/rmt/lib/rmt/checksum_verifier.rb:18:in `verify_checksum': Checksum doesn't match (RMT::ChecksumVerifier::Exception)
```

We should catch that and let RMT continue mirror other repos instead :ghost:

**Comment by ikapelyukhin** [comment]
Fixed by #193.
